fizz_buzz.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstdint>
+#include <unordered_map>
+
+enum class FizzBuzzResult : uint8_t {
+	UNKNOWN = 0, NONE = 1 << 0, FIZZ = 1 << 1, BUZZ = 1 << 2, FIZZBUZZ = FIZZ | BUZZ
+};
+
+struct _FizzBuzzer {
+	std::unordered_map<unsigned long long int, FizzBuzzResult> cache;
+	FizzBuzzResult fizzBuzz(const unsigned long long int number);
+};
+
+extern _FizzBuzzer fizzbuzzer;
```

Comment:
use inline instead of extern so you don't need to initialize the variable in the .cpp file

Reply:
Will inline fix the issue where initializing it directly in the header caused link errors? Can't test right now.

Reply:
`inline` means the same variable can be included multiple times across multiple translation units and still refer to the same variable. This doesn't actually mean the variable will be inlined. You can also use a local static variable so it's initialized on first call instead of before main():
```cpp
inline _FizzBuzzer& get_fizzbuzzer()
{
    static _FizzBuzzer fizzbuzzer;
    return fizzbuzzer;
}
```
might also want to use smart pointers idk

Reply:
Updated it to inline in commit f4550d9def95e0689b225d863f4e18e995daf1ab